diff_drive_controller/src/diff_drive_controller.cpp

```diff
@@ -374,9 +374,15 @@ namespace diff_drive_controller{
     config.publish_rate = publish_rate;
     config.enable_odom_tf = enable_odom_tf_;
 
-    dyn_reconf_server_ = std::make_shared<ReconfigureServer>(controller_nh);
+    dyn_reconf_server_ = boost::make_shared<ReconfigureServer>(mDynServerMutex, controller_nh);
+    ReconfigureServer::CallbackType f;
+    f = boost::bind(&DiffDriveController::reconfCallback, this, _1, _2);
+    dyn_reconf_server_->setCallback(f);
+    // Update parameters
+    dyn_reconf_server_->getConfigDefault(config);
```

Comment:
We don't want this `getConfigDefault()` - It will override the values we just set in `config`.

Reply:
I suspect this is why the tests failed.